hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,303 @@
+        """Core function for holographic reconstruction performing following steps:
+
+        * 2D FFT without apodisation;
+
+        * Cutting out sideband;
+
+        * Centering sideband;
+
+        * Applying round window;
+
+        * Applying sinc filter;
+
+        * Applying automatic filtering of Fresnel fringe (Fresnel filtering);
+
+        * Inverse FFT.
+
+        Parameters
+        ----------
+        holo_data : array_like
+            Holographic data array
+        sb_size : int
+            Size of the sideband filter in px
+        sb_pos : nparray (N=1)
+            Vector of two elements with sideband coordinates [y,x]
+        fresnel_ratio : float
+            The ratio of Fresnel filter with respect to the sideband size
+        fresnel_width : int
+            Width of fresnel fringe filter in px, set to 0 to disable filtering
+
+        Returns
+        -------
+            wav : nparray
+                Reconstructed electron wave
+
+        Notes
+        -----
+
+        Disabeling of Fresnel filter is not implemented
+
+        See Also
+        --------
+
+        reconstruct
+
+        """
+        # TODO: Parsing of the input has to be redone
+        # TODO: Add smoothing of Fresnel filter
+        # TODO: Add other smoothing options?
+
+        image_size = np.int(sb_size*2)
+        (sx, sy) = holo_data.shape
+        holo_data = np.float64(holo_data)
+
+        h_hw_fft = fftshift(fft2(holo_data))  # <---- NO Hanning filtering
+
+        sb_roi = h_hw_fft[(sb_pos[0]-image_size//2):(sb_pos[0]+image_size//2),
+                 (sb_pos[1]-image_size//2):(sb_pos[1]+image_size//2)]
+
+        (sb_ny, sb_nx) = sb_roi.shape
+        sb_l = min(sb_ny / 2, sb_nx / 2)
+        cen_yx = [sb_ny / 2, sb_nx / 2]
+
+        # Circular Aperture
+        cgrid_y = np.arange(-sb_ny / 2, sb_ny / 2, 1)
+        cgrid_x = np.arange(-sb_nx / 2, sb_nx / 2, 1)
+        (sb_xx, sb_yy) = np.meshgrid(cgrid_x, cgrid_y)
+        sb_r = np.sqrt(sb_xx ** 2 + sb_yy ** 2)
+        c_mask = np.zeros((sb_nx, sb_ny))  # Original: cMask=zeros(SB_Ny,SB_Nx);
+
+        c_mask[sb_r < sb_size] = 1
+
+        # Fresnel Mask
+        ang = np.arctan2((sx / 2 - sb_pos[0]), (sy / 2 - sb_pos[1]))  # [-pi pi]
+
+        p_one = np.round([fresnel_ratio * sb_l * np.sin(ang), fresnel_ratio * sb_l * np.cos(ang)])
+        p_two = np.round([sb_l * np.sin(ang), sb_l * np.cos(ang)])
+
+        ang_one = (ang - np.pi / 2) % (2 * np.pi )
+        ang_two = (ang + np.pi / 2) % (2 * np.pi )
+
+        r = fresnel_width / 2
+        aa = np.round([p_one[0] + r * np.sin(ang_one), p_one[1] + r * np.cos(ang_one)]) + cen_yx
+        bb = np.round([p_one[0] + r * np.sin(ang_two), p_one[1] + r * np.cos(ang_two)]) + cen_yx
+        cc = np.round([p_two[0] + r * np.sin(ang_two), p_two[1] + r * np.cos(ang_two)]) + cen_yx
+        dd = np.round([p_two[0] + r * np.sin(ang_one), p_two[1] + r * np.cos(ang_one)]) + cen_yx
```

Comment:
Due to broadcasting, `cen_yx` can be replaced by a simple `image_size` (`[image_size, image_size]` is not necessary).